### DISCUSSION TITLE
Cannot have `signin` in a URL path when sharing a Codespace

### DISCUSSION THREAD

**Discussion by jfrumar-infinitusai** [community/community#27968] [Codespaces] [answered]
We've encountered a quirky issue - we were constantly getting 400 errors we couldn't narrow down in our project when using a path like `https://<unique-codespace-url>/signin`:

![image](https://user-images.githubusercontent.com/53191729/182254162-6d1b1cf7-b0b2-4159-8fe0-098c59ba52b0.png)

Turns out, if we used `/signin.*/` in the URL path, we'd get this error (e.g. `/signin-two` also failed). Changing it to anything else fixes the issue. Our signin URL path is now `/login` to work around this issue.

It occurs when sharing the codespace port "publicly" or "private within org".

**Comment by sayhiben** [comment]
Do your web application logs show the request at all, or is the request terminating before that? (nginx/apache, web framework, etc.)

**Reply by jfrumar-infinitusai** [reply]
No - this appears to be an error from the Github web server, and the request never hits our application server.

**Reply by jeffwilcox** [reply]
+1 I hit the same. Definitely something in the GitHub proxy side, probably around how the auth is implemented.

In our app, it's gross, but for now, if we detect `process.env.CODESPACES`, at runtime our routes will listen on `/sign-in`, and then in prod and other environments, we keep it all `/signin`.

**Comment by thomas-sickert** [comment, marked as answer]
Thanks for the feedback. We're aware of this issue, and it's on our backlog to address it. Unfortunately we don't have a concrete timeline to share around the fix just yet. The current workaround is to use a path other than /signin*. We'll keep this thread updated as a fix is pushed out. Thanks for your patience.

**Reply by iRobot98** [reply]
it's July 2023 and still not sorted

**Reply by jeffwilcox** [reply]
@iRobot98 yeah I'm not proud but here is my team's current workaround https://github.com/search?q=repo%3Amicrosoft%2Fopensource-management-portal%20sign-in&type=code
